aptos-move/framework/src/module_metadata.rs

```diff
@@ -132,13 +135,21 @@ impl KnownAttribute {
     }
 }
 
+thread_local!(static METADATA_CACHE: RefCell<HashMap<Vec<Metadata>, Option<RuntimeModuleMetadataV1>>> = RefCell::new(HashMap::new()));
```

Comment:
agreed, hope we can avoid it in this PR

Reply:
I think the proper way to do this is to attach the derived data to the Module/Executable lifetime but probably require extra work

Reply:
This is a bit of disappointing, because the major ask we had already last fall to @gelash and to Dario at this point was to come up with an approach to cache derived information for loaded modules. Now we are saying this is postponed to some later work. In fact this PR REMOVES mostly previous caching (the type and function indices), so I'm suspicious you don't actually have any design for this in the first place.

Reply:
What do you mean by this? The metadata cache here is more of just for performance optimization that might be affected by this PR so it's an orthogonal issue. We definitely want to be able to cache the metadata alongside the resolved module but that's some future work. I tried a bit previously but it became really ugly.